base/task.jl

```diff
@@ -102,12 +147,16 @@ function task_done_hook(t::Task)
                 active_repl_backend.in_eval
                 throwto(active_repl_backend.backend_task, result)
             end
-            let bt = catch_backtrace()
-                # run a new task to print the error for us
-                @schedule with_output_color(:red, STDERR) do io
-                    print(io, "ERROR (unhandled task failure): ")
-                    showerror(io, result, bt)
-                    println(io)
+            if collect_ex(t)
+                t.result=CollectedException(result, catch_backtrace())
```

Comment:
This code only runs if `!handled`, i.e. if nobody calls `wait`, so I'm not sure this works for `@sync`.

I also believe there is a race condition here if the task exits before `COLLECT_TASK_EXCEPTIONS` can be set. Of course, there is already a race condition in the use of the `handled` condition to print the "unhandled task failure" message, but I wasn't so much in favor of that.

It would be better for task exceptions to contain their backtrace always or never, instead of controlling it with a task-local setting.